src/ds/thread_messaging.h

```diff
@@ -95,10 +76,40 @@ namespace threading
       } while (!item_head.compare_exchange_strong(tmp_head, item));
     }
 
+    struct TimerEntry
```

Comment:
The timer handling logic has expanded somewhat, and isn't completely trivial, it would be good to have some unit tests for it (there's a thread_messaging.cpp but it only has a single test to check no messages are leaked).

Reply:
#1758